src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -321,7 +343,11 @@ public void visitClassDef(JCClassDecl tree) {
      */
     @Override
     public void visitLambda(JCLambda tree) {
-        LambdaTranslationContext localContext = (LambdaTranslationContext)context;
+        LambdaTranslationContext localContext = new LambdaTranslationContext(tree);
+        if (dumpLambdaToMethodStats) {
```

Comment:
probably in a follow-up fix, but I don't see too much value on keeping / growing this stats code. This seems to belong in a regression test that probably uses a plugin or similar to get some stats while compiling lambdas / mrefs

Reply:
Actually... in an attempt to remove the translation context classes (and just keep the lambda translation context class - after all method references don't need much there), I stumbled upon a bad test failure in the lambda deduplication test. This test depends on the compiler generating the stat diagnostics for every encountered lambda... maybe we can drop that dependency and the stats, but probably better to do it elsewhere.